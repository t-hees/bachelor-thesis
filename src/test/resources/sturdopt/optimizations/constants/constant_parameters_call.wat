(module
  (type $func_type (func (param i32 i32 i32) (result i32)))
  (type $main_type (func (param i32) (result i32)))
  (func $add_func (type $func_type)
    local.get 0
    local.get 1
    i32.add
    local.get 2
    i32.add
  )
  (func $set_func (type $func_type)
    local.get 0
    local.tee 1 ;; This parameters is tee'd so it can't be removed anymore
    local.get 1
    i32.add
    local.get 2
    i32.add
  )
  (func (export "main") (type $main_type)
    local.get 0
    i32.const 4
    i32.const 5
    call $add_func

    local.get 0
    i32.const 4
    i32.const 5
    call $set_func

    i32.add
  )
)
