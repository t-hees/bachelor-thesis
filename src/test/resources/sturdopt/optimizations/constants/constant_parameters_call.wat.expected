(module
  (type $func_type (func (param i32 i32 i32) (result i32)))
  (type $main_type (func (param i32) (result i32)))
  (type $new_type (func (param i32 i32) (result i32)))
  (func $add_func (type $main_type)
    local.get 0
    i32.const 4
    i32.add
    i32.const 5
    i32.add
  )
  (func $set_func (type $new_type)
    i32.const 3
    local.tee 1 ;; This parameters is tee'd so it can't be removed anymore
    i32.const 3
    drop
    drop
    i32.const 6
    i32.const 5
    drop
    drop
    i32.const 11
  )
  (func (export "main") (type $main_type)
    local.get 0
    i32.const 4
    i32.const 5
    drop
    drop
    call $add_func

    i32.const 3
    i32.const 4
    i32.const 5
    drop
    drop
    drop
    i32.const 11

    i32.add
  )
)
